js/bitfinex2.js

```diff
@@ -410,34 +465,33 @@ module.exports = class bitfinex2 extends bitfinex {
             quoteId = this.getCurrencyId (quoteId);
             const minOrderSizeString = this.safeString (market, 'minimum_order_size');
             const maxOrderSizeString = this.safeString (market, 'maximum_order_size');
-            result.push ({
+            const parsedMarket = {
                 'id': 't' + id,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
-                'settle': undefined, // TODO
+                'settle': undefined,
                 'baseId': baseId,
                 'quoteId': quoteId,
-                'settleId': undefined, // TODO
-                'type': type,
+                'settleId': undefined,
+                'type': spot ? 'spot' : 'swap',
                 'spot': spot,
-                'margin': this.safeValue (market, 'margin'),
+                'margin': this.safeValue (market, 'margin', false),
                 'swap': swap,
                 'future': false,
                 'option': false,
                 'active': true,
                 'contract': swap,
-                'linear': spot ? undefined : true, // TODO
-                'inverse': spot ? undefined : false, // TODO
-                'contractSize': undefined, // TODO
-                'maintenanceMarginRate': undefined,
+                'linear': undefined,
+                'inverse': undefined,
+                'contractSize': undefined,
                 'expiry': undefined,
                 'expiryDatetime': undefined,
                 'strike': undefined,
                 'optionType': undefined,
                 'precision': {
-                    'amount': parseInt ('8'), // https://github.com/ccxt/ccxt/issues/7310
-                    'price': this.safeInteger (market, 'price_precision'),
+                    'amount': this.parsePrecision ('8'), // https://github.com/ccxt/ccxt/issues/7310
```

Comment:
`parsePrecision` returns strings, we should wrap it with `parseNumber`.

Reply:
In fact, Bitfinex2 inherits the SIGNIFICANT_DIGITS precision mode from Bitfinex1. So, a fraction won't work for precision, it requires whole numbers.

We should probably remove v1 and migrate to v2 entirely.

Reply:
> We should probably remove v1 and migrate to v2 entirely.

Not in this PR though, this is just to unify `fetchMarkets` which this is last PR required to do so